builtin/credential/approle/path_role.go

```diff
@@ -1314,17 +1368,38 @@ func (b *backend) pathRoleBoundCIDRListUpdate(ctx context.Context, req *logical.
 		return nil, nil
 	}
 
-	role.BoundCIDRList = data.Get("bound_cidr_list").([]string)
-	if len(role.BoundCIDRList) == 0 {
-		return logical.ErrorResponse("missing bound_cidr_list"), nil
+	if cidrListRaw, ok := data.GetOk("secret_id_bound_cidrs"); ok {
+		role.SecretIDBoundCIDRs = cidrListRaw.([]string)
+	} else if cidrListRaw, ok := data.GetOk("bound_cidr_list"); ok {
+		role.SecretIDBoundCIDRs = cidrListRaw.([]string)
 	}
 
-	valid, err := cidrutil.ValidateCIDRListSlice(role.BoundCIDRList)
-	if err != nil {
-		return nil, errwrap.Wrapf("failed to validate CIDR blocks: {{err}}", err)
+	if cidrListRaw, ok := data.GetOk("token_bound_cidrs"); ok {
```

Comment:
Same here.

Reply:
Same thing - if I do that and an update operation only changes _other_ fields, it wipes this out.

Reply:
Yep, keep as-is.